src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -55,6 +55,24 @@ public static bool IsSystemVoid([NotNullWhen(returnValue: true)] this ITypeSymbo
         public static bool IsNullable([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
             => symbol?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
 
+        public static bool IsNullable(
+            [NotNullWhen(returnValue: true)] this ITypeSymbol? symbol,
+            [NotNullWhen(returnValue: true)] out ITypeSymbol? underlyingType)
+        {
+            if (symbol is INamedTypeSymbol
+            {
+                OriginalDefinition: { SpecialType: SpecialType.System_Nullable_T },
+                TypeArguments: var args
+            })
+            {
+                underlyingType = args.Single();
+                return true;
+            }
+
+            underlyingType = null;
+            return false;
+        }
```

Comment:
Updated to fix the stack overflow. I was up too late last night. This is the interesting part, now used from `ShouldUseEqualityOperator`.

Reply:
This has now become (at @CyrusNajmabadi's request):

```cs
if (symbol is INamedTypeSymbol { OriginalDefinition: { SpecialType: SpecialType.System_Nullable_T } } namedType)
{
    underlyingType = namedType.TypeArguments.Single();
```